### PR TITLE
c2000-cgt: init at 22.6.1.LTS

### PR DESCRIPTION
**Pull request by teczito**
## Things done

Added a package with the Texas Instruments "C2000 Code Generation Tools"

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/c2/c2000-cgt/package.nix [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  lib,
+  stdenvNoCC,
+}:
+
+let
+  fhs = buildFHSEnv {
+    name = "fhs-shell";
+  };
+
+  package = stdenvNoCC.mkDerivation rec {
```
Comment (pluiedev) [inline pkgs/by-name/c2/c2000-cgt/package.nix:12]:
```suggestion
  package = stdenvNoCC.mkDerivation (finalAttrs: {
```

File: pkgs/by-name/c2/c2000-cgt/package.nix [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    pname = "cgt_c2000";
+    version = "22.6.1.LTS";
+
+    filename = "ti_${pname}_${version}_linux-x64_installer.bin";
+
+    src = builtins.fetchurl {
```
Comment (pluiedev) [inline pkgs/by-name/c2/c2000-cgt/package.nix:18]: Use pkgs.fetchurl

File: pkgs/by-name/c2/c2000-cgt/package.nix [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+      url = "https://dr-download.ti.com/software-development/ide-configuration-compiler-or-debugger/MD-xqxJ05PLfM/${version}/${filename}";
+      sha256 = "sha256:1nikmcgja2i22qk4alclfqiaqxwc7k0m9hwl5pviiv43bivb3rw7";
```
Comment (pluiedev) [inline pkgs/by-name/c2/c2000-cgt/package.nix:20]: Use SRI hashes (convert via `nix hash convert --hash-algo sha256`)
```suggestion
      hash = "sha256-h+exdlyD7Bj3LZTDVME8jHesInaUUUUmFiIKJR+rM9o=";
```

File: pkgs/by-name/c2/c2000-cgt/package.nix [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+  };
+
+  package = stdenvNoCC.mkDerivation rec {
+    pname = "cgt_c2000";
+    version = "22.6.1.LTS";
+
+    filename = "ti_${pname}_${version}_linux-x64_installer.bin";
+
+    src = builtins.fetchurl {
+      url = "https://dr-download.ti.com/software-development/ide-configuration-compiler-or-debugger/MD-xqxJ05PLfM/${version}/${filename}";
```
Comment (pluiedev) [inline pkgs/by-name/c2/c2000-cgt/package.nix:19]: Is the `MD-xqxJ05PLfM` part invariant of the version?

File: pkgs/by-name/c2/c2000-cgt/package.nix [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+      sha256 = "sha256:1nikmcgja2i22qk4alclfqiaqxwc7k0m9hwl5pviiv43bivb3rw7";
+    };
+
+    unpackPhase = "true";
```
Comment (pluiedev) [inline pkgs/by-name/c2/c2000-cgt/package.nix:23]:
```suggestion
    dontUnpack = true;
```

File: pkgs/by-name/c2/c2000-cgt/package.nix [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+    installPhase = ''
+      runHook preInstall
+
+      mkdir -p $out/install/
+      cp -a $src $out/install/installer.bin
```
Comment (pluiedev) [inline pkgs/by-name/c2/c2000-cgt/package.nix:29]: Does the installer really have to be in the output?

File: pkgs/by-name/c2/c2000-cgt/package.nix [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+
+      chmod +x $out/install/installer.bin
+      ${fhs}/bin/fhs-shell -c '$out/install/installer.bin --mode unattended --prefix /build/ti'
```
Comment (pluiedev) [inline pkgs/by-name/c2/c2000-cgt/package.nix:32]: I think you can just set the prefix to `$out` and then move everything outside the `ti-cgt-c2000_${version}` directory

File: pkgs/by-name/c2/c2000-cgt/package.nix [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+      mv /build/ti/ti-cgt-c2000_${version}/* $out/
+      rm -rf $out/install
+
+      runHook postInstall
+    '';
+
+    meta = with lib; {
+      description = "C2000-CGT - C28x/CLA code generation tools (CGT) - compiler";
```
Comment (pluiedev) [inline pkgs/by-name/c2/c2000-cgt/package.nix:40]:
```suggestion
      description = "C28x/CLA code generation tools";
```

File: pkgs/by-name/c2/c2000-cgt/package.nix [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+
+      chmod +x $out/install/installer.bin
+      ${fhs}/bin/fhs-shell -c '$out/install/installer.bin --mode unattended --prefix /build/ti'
+      mv /build/ti/ti-cgt-c2000_${version}/* $out/
+      rm -rf $out/install
+
+      runHook postInstall
+    '';
+
+    meta = with lib; {
```
Comment (pluiedev) [inline pkgs/by-name/c2/c2000-cgt/package.nix:39]: Avoid `with lib;`
```suggestion
    meta = {
```

File: pkgs/by-name/c2/c2000-cgt/package.nix [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+      homepage = "https://ti.org";
+      changelog = "https://software-dl.ti.com/codegen/esd/cgt_public_sw/C2000/${version}/README.html";
+      license = licenses.unfree;
+      maintainers = with lib.maintainers; [ teczito ];
+      mainProgram = "cl2000";
+      platforms = [ "x86_64-linux" ];
+    };
+  };
+in
+package
```
Comment (pluiedev) [inline pkgs/by-name/c2/c2000-cgt/package.nix:55]: You don't need this — just move the entire definition to the `in` part

File: pkgs/by-name/c2/c2000-cgt/package.nix [hunk 11]
```diff
@@ -0,0 +1,55 @@
+{
+  buildFHSEnv,
+  lib,
```
Comment (pluiedev) [inline pkgs/by-name/c2/c2000-cgt/package.nix:3]: By convention the order is lib first, then stdenv(NoCC), then everything else

**Review by pluiedev — CHANGES_REQUESTED** [review]
Welcome to Nixpkgs ❤️ Please squash your third commit into your second & change your PR title to `c2000-cgt: init at 22.6.1.LTS`

File: pkgs/by-name/c2/c2000-cgt/package.nix (hunk 4 above)
Comment (teczito) [reply pkgs/by-name/c2/c2000-cgt/package.nix:19]: Yes, it's a constant across all versions.

File: pkgs/by-name/c2/c2000-cgt/package.nix [hunk 12]
```diff
@@ -0,0 +1,51 @@
+{
+  lib,
+  stdenvNoCC,
+  buildFHSEnv,
+  pkgs,
```
Comment (pluiedev) [inline pkgs/by-name/c2/c2000-cgt/package.nix:5]:
```suggestion
  fetchurl,
```

File: pkgs/by-name/c2/c2000-cgt/package.nix [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+let
+  fhs = buildFHSEnv {
+    name = "fhs-shell";
+  };
+in
+stdenvNoCC.mkDerivation (finalAttrs: {
+  pname = "cgt_c2000";
+  version = "22.6.1.LTS";
+
+  src = pkgs.fetchurl {
```
Comment (pluiedev) [inline pkgs/by-name/c2/c2000-cgt/package.nix:17]:
```suggestion
  src = fetchurl {
```

**Review by pluiedev — CHANGES_REQUESTED** [review]
Changes still have to be squashed down to just two commits, btw

**Review by pluiedev — APPROVED** [review]

**Comment by pluiedev** [comment]
> @pluiedev Thank you so much for your kind and fast response! This is my first pull request. What should I do next?

Wait for someone with commit access to merge your PR (although, I'm fairly sure I missed something so it might take a while to address all concerns and merge)

**Review by wolfgangwalther — CHANGES_REQUESTED** [review]

**Review by wolfgangwalther — APPROVED** [review]

**Pull request merged**